app/controllers/api/v3/sync_controller.rb

```diff
@@ -35,6 +39,27 @@ def disable_audit_logs?
     false
   end
 
+  def log_block_level_sync_metrics(response_key)
```

Comment:
Centralize all the block-sync logging in one place so that the loggers don't end up being called multiple times during the code-path and are easy to find in one place.